app.js

```diff
@@ -41,19 +46,112 @@ app.get('/api/v1/projects/:id', async (req, res) => {
   }
 });
 
+
+// all palettes
+//'/api/v1/palettes'
+app.get('/api/v1/palettes', async (req, res) => {
+  try {
+    const { name } = req.query;
+    if (req.query.name) {
+      database('palettes')
+        .where('name', name)
+        .then(palette => {
+          if (palette) {
+            return res.status(200).json(palette);
+          } else {
+            return res.status(404).json('Palettes Not Found');
+          }
+        });
+    } else {
+      const palettes = await database('palettes').select();
+      if (!palettes.length) return res.status(404).json('Palettes Not Found');
+      res.status(200).json(palettes);
+    }
+  } catch (error) {
+    res.status(500).json({ error });
+  }
+});
+
```

Comment:
I'm wondering if we should be able to get all palettes associated with a single project as well....we'll have to write that logic somewhere in the app.